widgets/post-carousel/js/carousel.js

```diff
@@ -23,12 +23,20 @@ jQuery( function ( $ ) {
 				complete = numItems === totalPosts,
 				itemWidth = ( $firstItem.width() + parseInt( $firstItem.css( 'margin-right' ) ) ),
 				isRTL = $postsContainer.hasClass( 'js-rtl' ),
-				updateProp = isRTL ? 'margin-right' : 'margin-left';
+				updateProp = isRTL ? 'margin-right' : 'margin-left',
+				possibleVisibleItems = Math.ceil( $$.outerWidth() / itemWidth );
+
+				// Work out when to load new items based on number of posts on load and max number of visible items
+				// If numItems is less than 1, default to 1 to prevent issues with post loading
+				if ( numItems - possibleVisibleItems < 0 ) {
+					loadMoreTrigger = numItems - possibleVisibleItems;
+				} else {
+					loadMoreTrigger = 1;
+				}
```

Comment:
Currently this is fetching the next page of posts every time `updatePosition` is called. I think it's also a bit more complex than necessary as we already track the position of the carousel below.

Reply:
Yeah, you're right - I was over complicating this. 😞